src/kirby/components/calendar/helpers/calendar.helper.tns.ts

```diff
@@ -0,0 +1,127 @@
+// import 'reflect-metadata';
+
+import { ElementRef, NgZone } from '@angular/core';
+import { WebView, LoadEventData } from 'tns-core-modules/ui/web-view/web-view';
+import { isAndroid, isIOS } from 'tns-core-modules/platform';
+import { WebViewInterface } from 'nativescript-webview-interface';
+import * as applicationModule from 'tns-core-modules/application';
+
+import { CalendarOptions } from './calendar-options.model';
+
+declare const android;
+declare const UIColor;
+
+// declare var zonedCallback: Function;
+
+export class CalendarHelper {
+  private webViewInterface: WebViewInterface;
+  private webViewReady = false;
+
+  // constructor(private zone: NgZone) {}
+
+  public init(
+    calendarContainer: ElementRef,
+    options: CalendarOptions,
+    onDaySelected: (cell: { isSelectable: boolean; date: number }) => void,
+    onChangeMonth: (direction: number) => void
+  ) {
+    if (calendarContainer && calendarContainer.nativeElement) {
+      const webView = calendarContainer.nativeElement as WebView;
+      if (webView) {
+        webView.on(WebView.loadedEvent, (args: LoadEventData) => {
+          setTimeout(() => {
+            this.initializeWebView(webView, options, onDaySelected, onChangeMonth);
+          }, 0);
+        });
+      }
+    }
+  }
+
+  public update(options: CalendarOptions) {
+    this.emitOptionsToWebView(options);
+  }
+
+  public setSelectedDay(day: number) {
+    if (this.webViewReady) {
+      this.webViewInterface.emit('setSelectedDay', day);
+    }
+  }
+
+  private initializeWebView(
+    webView: WebView,
+    options: CalendarOptions,
+    onDaySelected: (cell: { isSelectable: boolean; date: number }) => void,
+    onChangeMonth: (direction: number) => void
+  ) {
+    if (webView) {
+      webView.on(WebView.loadFinishedEvent, (args: LoadEventData) => {
+        if (!args.error) {
+          this.webViewReady = true;
+          this.setupWebViewForPlatform(webView);
+          this.addEventListeners(onDaySelected, onChangeMonth);
+          this.emitOptionsToWebView(options);
+        }
+      });
+
+      this.webViewInterface = new WebViewInterface(webView, '~/calendar/calendar.webview.html');
+      this.setTransparentBackground(webView);
+    }
+  }
+
+  private emitOptionsToWebView(options: CalendarOptions) {
+    if (this.webViewReady) {
+      this.webViewInterface.emit('init', options);
+    }
+  }
+
+  private addEventListeners(
+    onDaySelected: (cell: { isSelectable: boolean; date: number }) => void,
+    onChangeMonth: (direction: number) => void
+  ) {
+    this.webViewInterface.on('daySelected', (selectedDay: number) => {
+      // TODO: Run in zone when DI bug regarding NgZone has been fixed:
```

Comment:
Consider adding a link to the Github issue, so that it's easier to track when the bug is fixed (not quite sure which bug is being referred to=

Reply:
Fixed